src/codegen/statements.rs

```diff
@@ -63,21 +63,76 @@ pub(crate) fn statement(
                     vartab,
                     opt,
                 };
-
                 //If we remove the assignment, we must keep expressions that have side effects
                 init.recurse(&mut params, process_side_effects_expressions);
                 return;
             }
 
             let expr = expression(init, cfg, contract_no, Some(func), ns, vartab, opt);
+
             cfg.add(
                 vartab,
                 Instr::Set {
                     loc: *loc,
                     res: *pos,
-                    expr,
+                    expr: expr.clone(),
                 },
             );
+
+            // Lets check if the declaration is a declaration of a dynamic array
+            if let Expression::AllocDynamicArray(_, _, size, _) = expr {
+                // If its a variable of uint32, assign it to a temp
+                if let Expression::Variable(_, ref ty, position) = *size {
+                    let num = Expression::Variable(pt::Loc::Codegen, Type::Uint(32), position);
+                    let temp_res = vartab.temp_name("array_size", ty);
+                    cfg.add(
+                        vartab,
+                        Instr::Set {
+                            loc: *loc,
+                            res: temp_res,
+                            expr: num.clone(),
+                        },
+                    );
+
+                    cfg.array_lengths_temps.insert(*pos, (temp_res, num));
+                }
+                // If size a uint and bits > 32
```

Comment:
Will we need to deal with an `uint8` or a `int16` as the size? If so, there would be a `SignExt` or a `ZeroExt`.

Reply:
@seanyoung

Reply:
I missed uint16, but is int16 possible as an array size?

Reply:
array size is always unsigned. Memory arrays are always 32 bit (may be 64 bit in the future on BPF).

Reply:
> array size is always unsigned. Memory arrays are always 32 bit (may be 64 bit in the future on BPF).

Yes, but I can create an array like this:
```
    function foo(uint8 v) public returns (int[] memory) {
        int[] vec = new int[](v);
        return vec;
    }
```
This implies a `ZeroExt` in the argument.

Reply:
If the function is like this:
``solidity
    function foo(int8 v) public returns (int[] memory) {
        int[] vec = new int[](uint32(v));
        return vec;
    }
```
Your argument will be a `SignExt`.

Reply:
This is now handled

Reply:
@LucasSte  @salaheldinsoliman  this does not need special handling.

```
    function foo(uint8 v) public returns (int[] memory) {
        int[] vec = new int[](v);
        return vec;
    }
```
In this code, sema will add `Expression::ZeroExt` to `v` already. So the length argument is `Type::Uint(32)`.